src/config/resolve-config-editorconfig.js

```diff
@@ -3,11 +3,9 @@
 const path = require("path");
 
 const editorconfig = require("editorconfig");
-const mem = require("mem");
 const editorConfigToPrettier = require("editorconfig-to-prettier");
 const findProjectRoot = require("./find-project-root.js");
-
-const jsonStringifyMem = (fn) => mem(fn, { cacheKey: JSON.stringify });
+const configCache = new Map();
```

Comment:
This changes the behavior, you are sharing cache between async and sync version, which is good, but I think it's not this simple.

The old async version not caching the result, but the promise, this will cause `Promise.all()` to search multiple times.

Reply:
i think i found it. wouldn't it mean a potential race condition more than anything? if we call it twice, quickly, neither will have cached yet and both will run whereas before it would've waited on one promise.

i can change to having two caches again if needed

Reply:
It's not about two caches, the async version need cache the promise, not result.

Reply:
yes, which means two caches in this case (we're keying by the same thing). or a tuple per cache entry (async / sync). unless you have a better suggestion?

Reply:
> yes, which means two caches in this case (we're keying by the same thing)

Wouldn't the `sync` flag be part of `opts` which we would be keying by in the cache, or am I mistaken here?